### PR TITLE
User name

### PR DESCRIPTION
**Pull request by garemoko**
Fetches the user's fullname using a standard Moodle function that takes into consideration all the international complexities of calculating a full name. 

This, combined with a PR I'm about to make to the translator should fix https://github.com/jlowe64/moodle-logstore_xapi/issues/60

Tests and doc updates are included. I also accidentally included the changes from https://github.com/LearningLocker/Moodle-Log-Expander/pull/31 but since that's about to be merged i guess no biggie? The relevant change is line 200 of repository.php

### CODE REVIEW

**Comment by ryansmith94** [comment]
Nice work putting it in the repo! That was the first thing I thought when I started reading your comment haha. Ping me in another comment here when you get the tests passing and I'll review it for Jerrett to merge.

**Comment by garemoko** [comment]
This is failing because the tests can't find the Moodle core function fullname, however this does work in the actual plugin. 

Any ideas on how to replicate these global Moodle functions in the tests @ryansmith94?

**Comment by garemoko** [comment]
@ryansmith94 guess i figured it out! Tests passing.

File: tests/EventTest.php [hunk 1]
```diff
@@ -1,101 +1,110 @@
...
-    protected $repo;
-
-    public function __construct() {
-        $this->cfg = (object) [
-            'wwwroot' => 'http://www.example.com',
-            'release' => '1.0.0',
-        ];
-        $this->repo = new TestRepository((object) [], $this->cfg);
+<?php 
+namespace {
```
Comment (ryansmith94) [inline tests/EventTest.php]: I don't like this, I don't know why, but it seems odd.

Comment (ryansmith94) [inline tests/EventTest.php]: In the actual repo I'd make a method on the repo class that does this call, then override that method in the test repo.

**Comment by ryansmith94** [comment]
All good besides the comment I just made on the code @garemoko

File: tests/EventTest.php (hunk 1 above)
Comment (garemoko) [inline tests/EventTest.php]: Got the approach from this blog: http://marcelog.github.io/articles/php_mock_global_functions_for_unit_tests_with_phpunit.html

It seems like unnecessary complexity to make additional methods to replace core Moodle functions? I guess it's what we've done with get_records though.

Comment (ryansmith94) [inline tests/EventTest.php]: I think having multiple namespaces in a single PHP file is probably frowned upon, so I don't want to take that route.

**Comment by garemoko** [comment]
@ryansmith94 your wish is my command :-) Done.

**Comment by ryansmith94** [comment]
Looks awesome, thanks @garemoko! :) ready to merge @jlowe64

**Pull request merged**